Sources/RudderStackAnalytics/Base/Analytics.swift

```diff
@@ -310,15 +349,45 @@ extension Analytics {
     }
     
     /**
-     Processes an event by dispatching it to the serial queue to ensure ordering without blocking the main thread.
+     Starts the background task that processes events from the channel through the plugin chain.
+     
+     This method creates a background task that continuously listens for events and processes them.
+     The task handles graceful shutdown and ensures proper cleanup when completed.
+     */
+    private func startProcessingEvents() {
+        
+        self.isAnalyticsShutdown = false
```

Comment:
As discussed offline, this might not be needed. @SKannaniOS will examine this further and remove it if not required.